test/test_sparse.py

```diff
@@ -1067,14 +1120,71 @@ def do_test(t):
+        x_v_numel = torch.zeros(x_v).numel()
+        y_v_numel = torch.zeros(y_v).numel()
+        x = torch.sparse_coo_tensor(torch.zeros(x_i),
+                                    torch.arange(x_v_numel).resize_(x_v).to(torch.float),
+                                    torch.Size(x_size))
+        x_dense = x.to_dense()
+        y = torch.sparse_coo_tensor(torch.zeros(y_i),
+                                    torch.ones(y_v).to(torch.float),
+                                    torch.Size(y_size))
+        y_dense = y.to_dense()
+        x.resize_as_(y)
+        x_dense.resize_as_(y_dense)
+        self.assertEqual(x.shape, y.shape)
+        self.assertEqual(x._sparseDims(), y._sparseDims())
+        self.assertEqual(x._denseDims(), y._denseDims())
+        # Here we make sure that the original data are preserved after resizing
+        self.assertEqual(x.to_dense().view(-1)[0:x_v_numel].view(x_v),
+                         x_dense.view(-1)[0:x_v_numel].view(x_v))
+
+    def test_resize(self):
+        # 1. Increase the size of some dense dimensions [Supported]
+        self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                [1, 1], [1, 2, 4], [2, 2, 4])
+
+        # 2. Expand the size of some sparse dimensions [Supported]
+        self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                [1, 1], [1, 2, 3], [4, 2, 3])
+
+        # 3. Change the shapes of both sparse and dense dimensions when nnz is zero [Supported]
+        self._test_resize_shape([1, 0], [0, 2, 3], [2, 2, 3],
+                                [2, 0], [0, 2, 4, 5], [1, 1, 2, 4, 5])
+
+        # 4. Add dims to dense dimensions [Not Supported]
+        with self.assertRaisesRegex(RuntimeError, "changing the number of dense dimensions"):
+            self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                    [1, 1], [1, 2, 3, 4], [2, 2, 3, 4])
+
+        # 5. Remove dims from dense dimensions [Not Supported]
+        with self.assertRaisesRegex(RuntimeError, "changing the number of dense dimensions"):
+            self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                    [1, 1], [1, 2], [2, 2])
+
+        # 6. Change the number of sparse dimensions on a non-empty sparse tensor [Not Supported]
+        with self.assertRaisesRegex(RuntimeError, "changing the number of sparse dimensions"):
+            self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                    [2, 1], [1, 2, 3], [1, 2, 2, 3])
+
+        # 7. Shrink the size of some sparse dimensions on a non-empty sparse tensor [Not Supported]
+        with self.assertRaisesRegex(RuntimeError, "shrinking the size of sparse dimensions"):
+            self._test_resize_shape([1, 1], [1, 2, 3], [2, 2, 3],
+                                    [1, 1], [1, 2, 3], [1, 2, 3])
+
+        # 8. Shrink the size of some dense dimensions on a non-empty sparse tensor [Not Supported]
```

Comment:
This case is unsupported now, because the following case gives different behavior for sparse and dense resize:

```
>>> x_i, x_v, x_size = [1, 1], [1, 2, 3], [2, 2, 3]
>>> y_i, y_v, y_size = [1, 1], [1, 2, 2], [2, 2, 2]
>>> x_v_numel = torch.zeros(x_v).numel()
>>> y_v_numel = torch.zeros(y_v).numel()
>>> x = torch.sparse_coo_tensor(torch.zeros(x_i), torch.arange(x_v_numel).resize_(x_v).to(torch.float), torch.Size(x_size))
>>> x_dense = x.to_dense()
>>> y = torch.sparse_coo_tensor(torch.zeros(y_i), torch.ones(y_v), torch.Size(y_size))
>>> y_dense = y.to_dense()
>>> x.resize_as_(y)
>>> print(x.to_dense())  # resize only affects the first block of elements
tensor([[[0., 1.],
         [2., 3.]],

         [[0., 0.],
          [0., 0.]]])
>>> x_dense.resize_as_(y_dense)
>>> print(x_dense)  # notice that the second block of elements is also affected
tensor([[[0., 1.],
         [2., 3.]],

        [[4., 5.],
        [0., 0.]]])
```